experimental/Schemes/CartierDivisor.jl

```diff
@@ -0,0 +1,40 @@
+export CartierDivisor
+export trivializing_covering
+
+@attributes mutable struct CartierDivisor{
```

Comment:
A potential problem: 
The way you model a Cartier divisor is basically as Weyl divisor + a cover where it is locally principal.
This has the drawback that we can only write down effective Cartier divisors and the group structure is not defined. Overcoming these obstructions seems awkward to me.
Defining them along the definition as global sections of $\mathcal{K}^*/\mathcal{O}^*$ seems more sustainable
in the long run.

Reply:
Sorry, this is too hard to me to imagine what your concerns are really about at this point. Let's discuss this in person.

Reply:
Alright, I think, I see better now what you mean. 

It is true: This class should probably be called `EffectiveCartierDivisor` rather than `CartierDivisor`. And I remember that I had already pondered about this renaming. In the Stacks project (linked above) they do the same thing: Define effective Cartier divisors and then extend this half group to an honest group by the usual process. 

So overcoming these difficulties is not that difficult in the end: A `CartierDivisor` should then simply be a formal sum of `EffectiveCartierDivisor`s, similar to our current implementation of `WeilDivisor`s. That would also have the benefit that we can absorb the exponents in the coefficients and don't bloat the local equations by exponentiation. 

The question is: Where do we want to go with this? What kind of functionality do we expect? The one I have in mind is intersection theory: I want to turn a `CartierDivisor` into a line bundle `L` and intersect any `WeylDivisor` with the first Chern class of `L`. For this, I need rational sections in a line bundle, right? The good thing is that the intersection is linear, so we can split the problem in taking line bundles of `EffectiveCartierDivisor`s and their rational sections and I think this can be brought to work. 

I don't have a clear idea of how to do things "along the definition as global sections of $$\mathcal K^\star \mathscr O^\star$$, but I see several problems there: We don't have the units of either $$\mathcal K^\star$$ nor $$\mathscr O^\star$$ as an honest group in Oscar. In particular, there is no `quo`-method that we could call on that. Maybe, we can fiddle our way around that, but that would still require introducing another sheaf (which is probably very hard to model) and sections in it. On a first glance, this seems like way more work to do it right.

Reply:
A Cartier divisor could be modeled by giving  $(U_i,f_i)$ where $f_i$ is an element of the total ring of fraction of $\mathcal{O}(U_i)$, the $U_i$ form an open cover of $X$ and $f_i/f_j | U_i \cap U_j$ is a unit in $\mathcal{O}$. 
I can see your concerns about this approach. 

With the theoretical backing of the stacks project, I can see your approach work too - and probably it is less work. 
Just give the references in the code & in the docu. So thumbs up.